central/authprovider/service/service_impl.go

```diff
@@ -165,8 +166,13 @@ func (s *serviceImpl) PostAuthProvider(ctx context.Context, request *v1.PostAuth
 	if providerReq.GetLoginUrl() != "" {
 		return nil, errox.InvalidArgs.CausedBy("auth provider loginUrl field is not empty")
 	}
+	if providerReq.GetRequiredAttributes() != nil {
+		return nil, errox.InvalidArgs.CausedBy("auth provider required attributes is set, this is not allowed " +
+			"for providers created via API.")
+	}
 
-	provider, err := s.registry.CreateProvider(ctx, authproviders.WithStorageView(providerReq), authproviders.WithValidateCallback(datastore.Singleton()))
+	provider, err := s.registry.CreateProvider(ctx, authproviders.WithStorageView(providerReq),
+		authproviders.WithValidateCallback(datastore.Singleton()), authproviders.WithAttributeChecker(providerReq))
```

Comment:
Seems like `authproviders.WithAttributeChecker(providerReq)` would not do anything

Reply:
Because we don't allow required attributes here?

Reply:
For the create case, I agree we can remove this right here, as in theory we won't allow creation via API with required_attributes set.

Reply:
I'm still not entirely convinced that we should not allow creation via API. Avoiding exceptions and special cases is a bliss!

Reply:
Clarified below, added the ability to create providers with `required_attributes` set once more.